buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -114,6 +116,17 @@ class BuildPlugin implements Plugin<Project> {
                 throw new GradleException("Java ${minimumJava} or above is required to build Elasticsearch")
             }
 
+            // TODO: remove when the build requires Java 9
+            if (javaVersionEnum.compareTo(JavaVersion.VERSION_1_8) == 0) {
+                if (Objects.equals("Oracle Corporation", javaVendor)) {
+                    def matcher = javaVersion =~ /1\.8\.0_(\d+)/
+                    int update = matcher[0][1].toInteger()
```

Comment:
can we be a bit more defensive here and make this dependent on matcher.matches()?